create contributor phone

turret-dev opened the pull request:
because contributors are epic 👍

SunburntRock89 commented:
There are 3 contributors. One doesn't use the bot at all. One of them is you. Perhaps you're trying to too your own horn slightly?

turret-dev commented:
but we have a redphone emote anyway so

SunburntRock89 commented:
You didn't deny it, and we don't need to use everything we have.

turret-dev commented:
i wouldn't even be able to use it because of my customer support so

turret-dev commented:
also according to (this)[https://github.com/DTel-HQ/dtel/graphs/contributors] we have 21 contributors including the dtel bosses and rexo

SunburntRock89 commented:
The people who have the role are the people who we care about here. If they don't have the role, they don't get the phone, and also probably don't care about the bot anymore

Rexogamer commented:
Sun did you fucking listen to Mitch's idea

Turret, I ❤️ this

Rexogamer commented on Internals/callHandler.js:
```diff
@@ -23,6 +23,7 @@ module.exports = async(cmd, msg, suffix, call) => {
 	let phone = config.callPhones.default;
 	if (fromvip) phone = config.callPhones.donator;
 	if (!hidden) {
+		if (msg.author.contributor) phone=config.callPhones.contributor;
```
Minor spacing error
```suggestion
		if (msg.author.contributor) phone = config.callPhones.contributor;
```

Mitchell3514 commented:
This wasn't exactly my idea. I thought the red phone was meant as a boss phone (which I also don't support, but don't mind either). 
Sun is definitely right, but also why not tbh.

SunburntRock89 commented:
Have you run this through ESLint? Also if you're gonna let this through, why don't you just go for the full shebang and make one for server boosters or something?

turret-dev commented:
boss phone would be like dark blue since their role color is blue

SunburntRock89 commented on Structures/DTelUser.js:
```diff
@@ -67,7 +68,7 @@ module.exports = Discord => {
 					manager: this.manager,
 					support: this.support,
-					donator: this.donator,
+					donator: this.donator
```
agreed

Rexogamer commented:
ideas:
- pink one for boosters
- normal blue phone but with like yellow stripes for boss?
- orange for supervisor (required ASAP)

SunburntRock89 commented:
And just like that, you've eliminated your chances of getting a supervisor badge

Rexogamer commented:
they were just ideas, who burnt you

austinhuang0131 commented:
boosters should not get badges because we don't get any benefits

turret-dev commented:
this is just a commit about making a red phone for contributors, not fucking giving every single person a unique phone

austinhuang0131 commented:
reminder that this is a public repo

no profanity

turret-dev commented:
shit i broke the rules fuck

turret-dev commented:
mitchell, fyi the reason i removed the comma is because i added a contributor perm but then i was like "nah, contributor only commands aint comin" so then i removed it and didnt know that eslint gets picky with commas so i sowwy

Rexogamer reviewed (approved):
back on topic, this lgtm code wise (although maybe the red phone should be boss instead?)

SunburntRock89 reviewed (changes requested):
Have you actually ran this and made sure it worked?

Rexogamer commented:
> Have you actually ran this and made sure it worked?

code wise this is basically the same as the other role phones, although whether it actually shows is another matter since CS is above it (out of scope but what if a user could have multiple phones showing)

SunburntRock89 commented:
> out of scope but what if a user could have multiple phones showing)
No, that'd just be cluttered

Mitchell3514 commented:
I'm not allowing us to lose our A+ on codacy, fix your code.

Rexogamer commented:
> I'm not allowing us to lose our A+ on codacy, fix your code.

all good now

Mitchell3514 commented:
This hound thing seems useless
